lib/alerts_viewer/application.ex

```diff
@@ -10,30 +10,31 @@ defmodule AlertsViewer.Application do
     load_runtime_config()
 
     children =
-      if Application.get_env(:alerts_viewer, :start_data_processes) do
-        [
-          Alerts.Supervisor,
-          Routes.Supervisor,
-          Vehicles.Supervisor,
-          TripUpdates.Supervisor,
-          TripUpdates.GTFSSupervisor
-        ]
-      else
-        []
-      end ++
-        [
-          Api.Cache,
-          # Start the Telemetry supervisor
-          AlertsViewerWeb.Telemetry,
-          # Start the PubSub system
-          {Phoenix.PubSub, name: AlertsViewer.PubSub},
-          # Start Finch
-          {Finch, name: AlertsViewer.Finch},
-          # Start the Endpoint (http/https)
-          AlertsViewerWeb.Endpoint
-          # Start a worker by calling: AlertsViewer.Worker.start_link(arg)
-          # {AlertsViewer.Worker, arg}
-        ]
+      [
+        Api.Cache,
+        # Start the Telemetry supervisor
+        AlertsViewerWeb.Telemetry,
+        # Start the PubSub system
+        {Phoenix.PubSub, name: AlertsViewer.PubSub},
+        # Start Finch
+        {Finch, name: AlertsViewer.Finch},
+        # Start the Endpoint (http/https)
+        AlertsViewerWeb.Endpoint
+        # Start a worker by calling: AlertsViewer.Worker.start_link(arg)
+        # {AlertsViewer.Worker, arg}
+      ] ++
```

Comment:
suggestion: I don't think we want to move these applications to the bottom of the children list. I think we want them all to be running before we start the web endpoint, otherwise a user could get an error because the data for a page they are requesting might not be available yet.

Reply:
I've pushed up a fix, it's a bit awkward because the snapshot logger needs to only start if start_data_processes are true (or it makes running test throw an error) but it has to be started after ApiCache. So for the fix I just put it in its own little thing.

Reply:
Ah, I see, I missed that nuance. This makes perfect sense if agreed, a bit awkward.